cli_tools/osconfig_agent/service/service.go

```diff
@@ -12,30 +12,34 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-// Package apipoller contains utilities to periodically poll the service for configurations.
-package apipoller
+// Package service runs the osconfig service.
+package service
 
 import (
 	"context"
+	"fmt"
 	"time"
 
-	osconfigAgent "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
-	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/tasker"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/config"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/inventory"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/logger"
-	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/osconfig"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/ospackage"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/patch"
 	"github.com/GoogleCloudPlatform/compute-image-tools/go/osinfo"
+	"github.com/GoogleCloudPlatform/compute-image-tools/go/service"
 	"github.com/kylelemons/godebug/pretty"
 	"google.golang.org/api/option"
+
+	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
 )
 
 var dump = &pretty.Config{IncludeUnexported: true}
 
-// Poll periodically calls the service to pull the latest applicaple configurations.
-func Poll(ctx context.Context) {
-	client, err := osconfigAgent.NewClient(ctx, option.WithEndpoint(config.SvcEndpoint()), option.WithCredentialsFile(config.OAuthPath()))
+func run(ctx context.Context) {
+	client, err := osconfig.NewClient(ctx, option.WithEndpoint(config.SvcEndpoint()), option.WithCredentialsFile(config.OAuthPath()))
```

Comment:
Not needed for this change but something I was wondering, can we move this to inside the for loop? That way we reread the endpoint and credentials for each call which allows for them to change w/out restarting the service. It can make it easier to fix/debug problems.

Reply:
That makes sense especially if we want configs to change without a service restart, I'll work on that for the next PR